src/solver/ts-generator/include/antares/solver/ts-generator/generator.h

```diff
@@ -41,11 +41,18 @@ template<enum Data::TimeSeriesType T>
 bool GenerateTimeSeries(Data::Study& study, uint year, IResultWriter& writer);
 
 bool GenerateThermalTimeSeries(Data::Study& study,
+                               std::vector<Data::ThermalCluster*> clusters,
                                uint year,
-                               bool globalThermalTSgeneration,
-                               bool refresh,
-                               IResultWriter& writer);
+                               Solver::IResultWriter& writer);
 
+std::vector<std::pair<std::string, std::string>> splitStringIntoPairs (const std::string& s,
```

Comment:
**splitStringIntoPairs** returns a vector of pairs. It could return a **std::map** instead, more natural.
There are probably a reason for that.
**Question** : why do we not return a **std::map** ?

Reply:
we can have [area1, cluster1] [area1, cluster2], map won't work

Reply:
A **std:map** is nothing but a vector of pairs (more or less).
But you're right : with a **std::map**, we can have identical keys (area names), unless we make the cluster names the keys, which can be identical.
The **std::map** does not seem so natural now.